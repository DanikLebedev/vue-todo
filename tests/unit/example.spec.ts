import { mount, shallowMount } from '@vue/test-utils';
import Page404 from '@/views/404Page.vue';
import TodoItem from '@/components/todo-item.vue';
import dynamicComponent from '@/components/dynamic-component.vue';
import CompositionApiComponent from '@/components/composition_api_component.vue';

describe('Render Pages', () => {
  it('renders 404 page', () => {
    const wrapper = mount(Page404);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('render correct todo item', () => {
    const todo = shallowMount(TodoItem, {
      props: {
        todo: { id: 2, title: 'test', done: false }
      }
    });
    expect(todo.element).toMatchSnapshot();
  });

  it('render dynamic components', () => {
    const dynamicComponents = shallowMount(dynamicComponent);
    expect(dynamicComponents.element).toMatchSnapshot();
  });

  it('render composition api component', () => {
    const dynamicComponents = mount(CompositionApiComponent);
    expect(dynamicComponents.html()).toMatchSnapshot();
  });
});
