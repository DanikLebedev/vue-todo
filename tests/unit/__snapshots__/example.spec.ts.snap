// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Pages render composition api component 1`] = `
<div class="">
  <h1>Composition API</h1>
  <div>
    <h2>Counter via state</h2><button class="p-button p-component" type="button">+</button><button class="p-button p-component ml-1" type="button">-</button>
    <p>Counter 0</p>
    <h2>Counter via ref</h2><button class="p-button p-component" type="button">+</button><button class="p-button p-component ml-1" type="button">-</button>
    <p>Counter 0</p>
  </div>
</div>
`;

exports[`Render Pages render correct todo item 1`] = `
<div
  class="p-card p-shadow-3"
>
  <div
    class="p-text-secondary done-text"
  >
    test
  </div>
  <div
    class="buttons-wrapper"
  >
    <button-stub />
    <button-stub
      class="p-button-danger"
    />
  </div>
</div>
`;

exports[`Render Pages render dynamic components 1`] = `
<div
  class="dynamic-component-wrapper"
/>
`;

exports[`Render Pages renders 404 page 1`] = `
<h1>
  Ooooops, Page not found
</h1>
`;
